### PR TITLE
Add MariaDB notes to docs

### LINKED ISSUE #3186: Document how to use MariaDB with SCDF+Skipper
The user would have explicitly update the JDBC URL with `--spring.datasource.url='jdbc:mariadb://localhost:3306/dataflow?useMysqlMetadata=true'`.

### PR DESCRIPTION
**Pull request by jvalkeal**
- Add notes about connector 2.4.1 new useMysqlMetadata
  option and example how to define datasource targeting
  MariaDB.
- Fixes #3186

### CODE REVIEW

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration-local.adoc [hunk 1]
```diff
@@ -62,6 +62,21 @@ java -jar spring-cloud-dataflow-server/target/spring-cloud-dataflow-server-{proj
     --spring.datasource.driver-class-name=org.mariadb.jdbc.Driver &
 ----
 
+The following example shows how to define a MariaDB database connection with command Line arguments
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration-local.adoc]: Perhaps: 

> The following example shows how to configure MariaDB database connection with command-line arguments

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration-local.adoc [hunk 2]
```diff
@@ -62,6 +62,21 @@ java -jar spring-cloud-dataflow-server/target/spring-cloud-dataflow-server-{proj
...
+[source,bash,subs=attributes]
+----
+java -jar spring-cloud-dataflow-server/target/spring-cloud-dataflow-server-{project-version}.jar \
+    --spring.datasource.url=jdbc:mariadb://localhost:3306/mydb?useMysqlMetadata=true \
+    --spring.datasource.username=<user> \
+    --spring.datasource.password=<password> \
+    --spring.datasource.driver-class-name=org.mariadb.jdbc.Driver &
+----
+
+NOTE: Starting with MariaDB connector 2.4.1, it is required to add option `useMysqlMetadata=true`
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration-local.adoc]: > Starting with MariaDB v2.4.1 connector release, it is required to also add `useMysqlMetadata=true` to the JDBC URL. This is a required workaround until when MySQL and MariaDB entirely switch as two different databases.

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration-local.adoc (hunk 1 above)
Comment (jvalkeal) [reply spring-cloud-dataflow-docs/src/main/asciidoc/configuration-local.adoc]: I copy pasted other sections and just changed db name, so keeping it consistent. Not gonna change.

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration-local.adoc (hunk 2 above)
Comment (jvalkeal) [reply spring-cloud-dataflow-docs/src/main/asciidoc/configuration-local.adoc]: Yeah, this is fine.

**Comment by jvalkeal** [comment]
Merged per b77728a50ebb7f36c725cec12dcabc462f74dce9

**Pull request closed without merging**